coil-base/src/main/java/coil/ImageLoader.kt

```diff
@@ -63,6 +64,7 @@ interface ImageLoader {
     /**
      * Completely clear this image loader's memory cache and bitmap pool.
      */
+    @MainThread
     fun clearMemory()
```

Comment:
These functions have always called through to data structures that aren't thread safe, however now I'm making this assumption explicit. We could support calling these from any thread, however it would slightly slow down every image request due to synchronization so let's optimize for the common case.

You can technically still call these methods off the main thread, however it's not 100% safe due to race conditions.